operator/pkg/manifest/helm_processor.go

```diff
@@ -437,54 +615,40 @@ func (h *helm) GetNsResource() (kube.ResourceList, error) {
 }
 
 func (h *helm) getTargetResources(ctx context.Context, manifest string,
-	transforms []types.ObjectTransform, object types.BaseCustomObject,
+	transforms []types.ObjectTransform, object types.BaseCustomObject, retryOnNoMatch bool,
 ) (kube.ResourceList, error) {
-	var resourceList kube.ResourceList
-	var err error
-
-	if len(transforms) == 0 {
-		resourceList, err = h.clients.KubeClient().Build(bytes.NewBufferString(manifest), false)
-	} else {
-		resourceList, err = h.transformManifestResources(ctx, manifest, transforms, object)
-	}
-
-	if err != nil {
-		//nolint:errorlint
-		errAggregate, ok := err.(apiMachineryErr.Aggregate)
-		if ok {
-			wrappedErr := resourceValidationErr
-			for _, nestedErr := range errAggregate.Errors() {
-				wrappedErr = fmt.Sprintln(wrappedErr, nestedErr.Error())
-			}
-			return nil, fmt.Errorf(wrappedErr)
-		}
-		return nil, fmt.Errorf("%s: %w", resourceValidationErr, err)
-	}
+	resourceList, resourceListErr := h.resourceListFromManifest(ctx, manifest, transforms, object, retryOnNoMatch)
 
 	// verify namespace override if not done by kubeclient
-	if err = overrideNamespace(resourceList, h.clients.Install().Namespace); err != nil {
-		return nil, err
+	if namespaceErr := overrideNamespace(resourceList, h.clients.Install().Namespace); namespaceErr != nil {
+		return resourceList, namespaceErr
```

Comment:
We should still return nil here with the error. Since resources without overridden namespace are incorrect. 

Also a minor thing - we can use the same `err` instead of `namespaceErr`